### PR TITLE
completed challenges

### PR DESCRIPTION
**Pull request by dnlclmnt**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,39 +14,257 @@
...
+//   ✕ takes in a string and returns a coded message. The coded message converts 'a' to 4 'e' to 3, 'i' to 1, and 'o' to 0. (1 ms)
+
+// ● codedMsg › takes in a string and returns a coded message. The coded message converts 'a' to 4 'e' to 3, 'i' to 1, and 'o' to 0.
+
+//   ReferenceError: codedMsg is not defined
+
 
 // b) Create the function that makes the test pass.
 
+// Create function that takes in a string.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:44]: this pseudocode looks good

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,39 +14,257 @@
...
+        }
+        else {
+            newStr += str[i]
+        }
+    }
+    return newStr
+}
+
+
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:81]: Refactor idea: look for a tool that will perform that logic to reassign the values

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,39 +14,257 @@
...
+        // (Test for edge cases of capitalized letters by calling on the .toLowerCase() method on the letter we're comparing 
+        // as well as (logical AND operator) making sure that the word is not already in our result array (use indexOf method to check that the word we are iterating over isn't already in our result array)) 
+        // then...
+            // push word into result array
+// return result array
+
+// In our function, arr[i] is representative of the current word in the array
+    // We are using a nested for loop to also iterate over each word to compare each letter in the word (represented as arr[i][j]) with the letter parameter we are passing into our function
+
+const certainLetter = (arr, letter) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:137]: very creative. same suggestion for the refactor idea. I only suggest refactors because you have grasped the concept behind these tools so let's see if there are other useful tools for you.

File: code-challenges.test.js [hunk 4]
```diff
@@ -14,39 +14,257 @@
...
+//     const handArr = Object.values(hand).sort()
+//     for (let i = 0; i < handArr.length; i++) {
+//         if (handArr[i] === 2 && handArr[i + 1] === 3) {
+//             return true
+//         } else {
+//             return false
+//         }
+//     }
+// }
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:253]: like that you took some different approaches

File: interview-questions.md [hunk 5]
```diff
@@ -8,44 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: The website rubygems.org has tons of community-shared gems that also show how you can add the gems to your project with documented install commands. RSpec is an example of a gem on this website. Navigating the site, we can learn to add the RSpec gem to our project using the $ gem install rspec command in our terminal.
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is an open-source framework that combines the Ruby language with the Rails framework and allows a user to quickly and easily create full-stack applications. Though my experience with it has been limited, it has allowed me to see how the back-end of a web application can be modified, updated, and viewed.
 
-Researched answer:
+Researched answer: Ruby on Rails utilizes a Model-View-Controller architectural pattern. The model is the structure of the application and manages the data and logic of the app. The view is the visual representation of the app. The controller receives user input and interacts with data model objects.
```
Comment (SunkissedQueen) [inline interview-questions.md:29]: I think some examples of each (MVC) would make this answer stand out. For instance you can speak on a website that you use a lot.

File: interview-questions.md [hunk 6]
```diff
@@ -8,44 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: Ruby on Rails utilizes a Model-View-Controller architectural pattern. The model is the structure of the application and manages the data and logic of the app. The view is the visual representation of the app. The controller receives user input and interacts with data model objects. 
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is a database of items that share a relation with other items within the database. An example of this would be when creating a table of books and another table of authors. They would share one item in common, a foreign key, to reference the primary key from the other table. This would be how we could link several book titles to one author.
 
-Researched answer:
+Researched answer: There are non-relational databases that have limited functionality and have limited indexing. Some examples of sites that use relational databases are YouTube and Twitter. Facebook, Amazon, Netflix, and Google are some examples of sites that use non-relational databases.
```
Comment (SunkissedQueen) [inline interview-questions.md:35]: I would suggest stating appropriate circumstances for each type of database.

File: interview-questions.md [hunk 7]
```diff
@@ -8,44 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are unique values given to items in a database. They are important because they can be used as the link between tables in a relational database. In my experience using Rails, I've created one table of users, then I've created a separate table of credits that are linked to individual users due to their specific primary key.
```
Comment (SunkissedQueen) [inline interview-questions.md:40]: Your example would be a good way to show that you understand when associating tables with the primary key of a different table it is called a foreign key.

File: ruby-challenges.rb [hunk 8]
```diff
@@ -10,23 +10,96 @@
...
+# Use the select method (works like the filter method in JS) on the array parameter
+    # if the word in the array contains the letter argument, it will be returned in a new array
+
+
+def words_with_letter(arr, letter)
+    arr.select do |word|
+        word.include? letter
+    end
+end
+
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:24]: Where's the method call?

File: ruby-challenges.rb [hunk 9]
```diff
@@ -10,23 +10,96 @@
...
+# Use flatten method on container to remove nested arrays and sort method to sort our single array into alphabetical order
+
+def noHash hash
+    container = []
+    hash.each do |key, value|
+        container << value
+    end
+    p container.flatten!.sort!
+end
+
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:46]: Remember to perform the method call with the test variables

File: ruby-challenges.rb [hunk 10]
```diff
@@ -10,23 +10,96 @@
...
+my_bike = Bike.new
+my_bike.bike_info
+my_bike.pedal_faster(10)
+my_bike.bike_info
+my_bike.pedal_faster(18)
+my_bike.bike_info
+my_bike.brake(5)
+my_bike.bike_info
+my_bike.brake(25)
+my_bike.bike_info
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:105]: What would you add to this class to allow the model and wheels value to be modified?

**Review by SunkissedQueen — APPROVED** [review]
Daniel, you have certainly given some in depth pseudocode. Appreciate your creative approach to the functions. You got your tests to pass!! I would definitely suggest using some high order functions to carry some of that load for your logic. For the ruby challenges please make sure to include the method call. Your interview answers are nice. I left some notes. This assessment is approved.
